### PR TITLE
{devel}[intel/2017b] SCons v3.0.1

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2410.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/68e22229e26af49944279776d0880afd for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2085.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/85e165d8a2da0cec30e09d1f629d8e24 for a full test report.

File: easybuild/easyconfigs/s/SCons/SCons-3.0.1-intel-2017b-Python-3.6.3.eb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+name = 'SCons'
+version = '3.0.1'
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'http://www.scons.org/'
+description = "SCons is a software construction tool."
+
+toolchain = {'name': 'intel', 'version': '2017b'}
+
+source_urls = [PYPI_SOURCE]
```
Comment (damianam) [inline easybuild/easyconfigs/s/SCons/SCons-3.0.1-intel-2017b-Python-3.6.3.eb]: It is also available in sourceforge, and previous easyconfigs used that URL. I am not arguing about switching to pypi, I am just wondering the rationale behind

File: easybuild/easyconfigs/s/SCons/SCons-3.0.1-intel-2017b-Python-3.6.3.eb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+sources = [SOURCELOWER_TAR_GZ]
+checksums = ['24475e38d39c19683bc88054524df018fe6949d70fbd4c69e298d39a0269f173']
+
+dependencies = [('Python', '3.6.3')]
```
Comment (damianam) [inline easybuild/easyconfigs/s/SCons/SCons-3.0.1-intel-2017b-Python-3.6.3.eb]: Previous easyconfigs didn't need a Python package (the one from the OS was used). Maybe comment here why this change? I suppose SCons 3 needs Python 3

File: easybuild/easyconfigs/s/SCons/SCons-3.0.1-intel-2017b-Python-3.6.3.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/s/SCons/SCons-3.0.1-intel-2017b-Python-3.6.3.eb]: I based this on an existing easyconfig which is already using `PYPI_SOURCE`:

```
$ grepi PYPI easybuild/easyconfigs/s/SCons/*eb
easybuild/easyconfigs/s/SCons/SCons-2.5.1-intel-2017a-Python-2.7.13.eb:12:source_urls = [PYPI_SOURCE]
```

Not sure which I prefer, but I guess consistently would be nice, yes.

I'll look into it.

File: easybuild/easyconfigs/s/SCons/SCons-3.0.1-intel-2017b-Python-3.6.3.eb (hunk 2 above)
Comment (boegel) [reply easybuild/easyconfigs/s/SCons/SCons-3.0.1-intel-2017b-Python-3.6.3.eb]: We have one version that relies on the system Python, because we wanted it to use in combination with Python 3 while `SCons` 2.x doesn't actually support Python 3 yet (so we called `scons` using the `python2` from the OS).

This was a bit hackish, and with `SCons` 3.x that is no longer needed, since they now properly support both Python 2.7 and 3.x, so no need to rely on the system Python anymore.

File: easybuild/easyconfigs/s/SCons/SCons-3.0.1-intel-2017b-Python-3.6.3.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/s/SCons/SCons-3.0.1-intel-2017b-Python-3.6.3.eb]: Updated to use SourceForge for consistency, checksums are not affected.

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
rescomp2 - Linux centos 6.8, Intel(R) Xeon(R) CPU E5-2650 v2 @ 2.60GHz, Python 2.7.14
See https://gist.github.com/3239765ec5ab23ad69d55faf6de59413 for a full test report.

**Comment by verdurin** [comment]
Going in, thanks @boegel!
